flavors/lineageos/default.nix

```diff
@@ -115,26 +122,29 @@ in mkIf (config.flavor == "lineageos")
       # So we'll just build chromium webview ourselves.
       "external/chromium-webview".enable = false;
     }
-  ] ++ optionals (deviceMetadata ? "${config.device}") [
+  ] ++ optionals (deviceMetadata ? "${config.device}") (let
     # Device-specific source dirs
-    (let
-      vendor = toLower deviceMetadata.${config.device}.vendor;
-      relpathWithDependencies = relpath: [ relpath ] ++ (flatten (map (p: relpathWithDependencies p) deviceDirs.${relpath}.deps));
-      relpaths = relpathWithDependencies "device/${vendor}/${config.device}";
-      filteredRelpaths = remove (attrNames repoDirs) relpaths; # Remove any repos that we're already including from repo json
-    in filterDirsAttrs (getAttrs filteredRelpaths deviceDirs))
-
+    vendor = toLower deviceMetadata.${config.device}.vendor;
+    relpathWithDependencies = relpath: [ relpath ] ++ (flatten (map (p: relpathWithDependencies p) deviceDirs.${relpath}.deps));
+    relpaths = relpathWithDependencies "device/${vendor}/${config.device}";
+    filteredRelpaths = remove (attrNames repoDirs) relpaths; # Remove any repos that we're already including from repo json
+
+    # Device-specific common vendor dirs
+    deviceDirsCommon = filter (path: hasPrefix "device/" path && hasSuffix "-common" path && !hasSuffix "xiaomi/sm8350-common" path) relpaths;
+    vendorDeviceDirs = map (replaceStrings [ "device/" ] [ "vendor/" ]) deviceDirsCommon;
+  in [
+    (filterDirsAttrs (getAttrs filteredRelpaths deviceDirs))
+    (filterDirsAttrs (getAttrs vendorDeviceDirs vendorDirs))
+  ]) ++ [
     # Vendor-specific source dirs
     (let
-      _vendor = toLower deviceMetadata.${config.device}.vendor;
-      vendor = if config.device == "shamu" then "motorola" else _vendor;
+      vendor = if config.androidVersion >= 13 then
+        deviceMetadata.${config.device}.vendor_dir_new
+      else
+        deviceMetadata.${config.device}.vendor_dir;
       relpath = "vendor/${vendor}";
     in filterDirsAttrs (getAttrs [relpath] vendorDirs))
-  ] ++ optional (config.device == "bacon")
-    # Bacon needs vendor/oppo in addition to vendor/oneplus
-    # See https://github.com/danielfullmer/robotnix/issues/26
-    (filterDirsAttrs (getAttrs ["vendor/oppo"] vendorDirs))
-  );
```

Comment:
There's a bug in this breaking los19. These changes were kind of rushed.

I'll try and fix it but might just drop los 19 since I don't use it myself.